packages/aragon-wrapper/src/index.js

```diff
@@ -454,6 +458,110 @@ export default class Aragon {
     this.forwarders.connect()
   }
 
+  /**
+   * Initialise identity providers.
+   *
+   * @return {Promise<void>}
+   */
+  async initIdentityProviders () {
+    const defaultIdentityProviders = [{
+      name: 'local',
+      provider: new LocalIdentityProvider()
+    }]
+    // TODO: detect other installed providers
+    const detectedIdentityProviders = []
+    const identityProviders = [...defaultIdentityProviders, ...detectedIdentityProviders]
+
+    // Init all providers
+    await Promise.all(identityProviders.map(({ provider }) => {
+      // Most providers should have this defined to a noop function by default, but just in case
+      if (typeof provider.init === 'function') {
+        return provider.init()
+      }
+      return Promise.resolve()
+    }))
+
+    this.identityProviderRegistrar = new Map(
+      identityProviders.map(({ name, provider }) => [name, provider])
+    )
+    // Set up identity modification intent observable
+    this.identityIntents = new Subject()
+  }
+
+  /**
+   * Modify the identity metadata for an address using the highest priority provider.
+   *
+   * @param  {string} address Address to modify
+   * @param  {Object} metadata Modification metadata object
+   * @return {Promise}
+   */
+  modifyAddressIdentity (address, metadata) {
+    const providerName = 'local'
+    const provider = this.identityProviderRegistrar.get(providerName)
+    if (provider && typeof provider.modify === 'function') {
+      return provider.modify(address, metadata)
+    }
+    throw new Error(`Provider (${providerName}) not installed`)
+  }
+
+  /**
+   * Resolve the identity metadata for an address using the highest priority provider.
+   *
+   * @param  {string} address Address to resolve
+   * @return {Promise}
+   */
+  resolveAddressIdentity (address) {
+    const providerName = 'local' // TODO - get provider
+    const provider = this.identityProviderRegistrar.get(providerName)
+    if (provider && typeof provider.resolve === 'function') {
+      return provider.resolve(address)
+    }
+    throw new Error(`Provider (${providerName}) not installed`)
```

Comment:
Ahh, we should be careful here and in `modifyAddressIdentity()`; it probably makes sense for the API to return a rejected promise instead given the async nature of these functions.